test/libsolidity/syntaxTests/abiEncoder/encodeCallMismatch.sol

```diff
@@ -0,0 +1,45 @@
+
+contract C {
+	function f(int a) public {}
```

Comment:
I'd put this test in `syntaxTests/specialFunctions/encodeCall.sol`. That's where tests for other `abi.*` functions seem to be.